crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1540,38 +1554,68 @@ impl CumulocityConverter {
             .and_then(|entity| C8yTopic::smartrest_response_topic(entity, &self.config.c8y_prefix))
             .ok_or_else(|| Error::UnknownEntity(target.to_string()))?;
 
-        match response.status() {
+        let messages = match command.status() {
             CommandStatus::Init | CommandStatus::Scheduled | CommandStatus::Unknown => {
                 // The command has not been processed yet
-                Ok(vec![])
+                vec![]
             }
             CommandStatus::Executing => {
                 let smartrest_set_operation_status =
                     set_operation_executing(CumulocitySupportedOperations::C8ySoftwareUpdate);
-                Ok(vec![MqttMessage::new(
-                    &topic,
-                    smartrest_set_operation_status,
-                )])
+                vec![MqttMessage::new(&topic, smartrest_set_operation_status)]
             }
             CommandStatus::Successful => {
                 let smartrest_set_operation =
                     succeed_operation_no_payload(CumulocitySupportedOperations::C8ySoftwareUpdate);
 
-                Ok(vec![
+                vec![
                     MqttMessage::new(&topic, smartrest_set_operation),
-                    response.clearing_message(&self.mqtt_schema),
+                    command.clearing_message(&self.mqtt_schema),
                     self.request_software_list(target),
-                ])
+                ]
             }
             CommandStatus::Failed { reason } => {
                 let smartrest_set_operation =
                     fail_operation(CumulocitySupportedOperations::C8ySoftwareUpdate, &reason);
 
-                Ok(vec![
+                vec![
                     MqttMessage::new(&topic, smartrest_set_operation),
-                    response.clearing_message(&self.mqtt_schema),
+                    command.clearing_message(&self.mqtt_schema),
                     self.request_software_list(target),
-                ])
+                ]
+            }
+        };
+
+        Ok((messages, Some(command.into())))
+    }
+
+    async fn upload_operation_log(&mut self, command: GenericCommand) {
+        if command.target == *self.entity_store.main_device()
```

Comment:
Don't we have to also upload the log file for child devices?

Reply:
Yes, I'll revert that. This guard was added in the initial version, where we were uploading the software plugin log, instead of the workflow log, which was only available for the main device. But now that we've switched to uploading the workflow logs, this guard is not needed anymore.